### ISSUE TITLE
Having trouble with GCE cloud profile assigning static IP and enabling IP forward

### ISSUE THREAD

**Issue by fyatzeck** [saltstack/salt#21390]
Hello,
My first issue when creating a GCE instance is I want to specify a static IP as the external IP using a pre-defined static IP as below:
```
external_ip: staticip-perf1-ip1
```
Unfortunately, Salt Cloud always provisions the instance with an Ephemeral IP.  I also tried with quotes around "staticip-perf1-ip1" but same result.
Work around is to manually go into gcloud console and re-assign static IP after instance is created.  But it would be nice to have instance start with static IP initially assigned, is this supported?

The second issue, somewhat unrelated, is there a cloud profile config setting to enable IP forwarding on GCE instances?  We want some instances to be provisioned as a NAT, I tried adding "canIpForward: True" but no luck.

Thanks.

**Comment by fyatzeck** [comment]
Here are my versions:
salt --versions-report
    Salt: 2014.7.1
    Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
    Jinja2: 2.7.2
    M2Crypto: 0.21.1
    msgpack-python: 0.3.0
    msgpack-pure: Not Installed
    pycrypto: 2.6.1
    libnacl: Not Installed
    PyYAML: 3.10
    ioflo: Not Installed
    PyZMQ: 14.0.1
    RAET: Not Installed
    ZMQ: 4.0.4
    Mako: 0.9.1

**Comment by erjohnso** [comment]
Thanks for reporting this @fyatzeck. You've bumped up against a couple missing features in the GCE driver.  Fortunately, this functionality does exist in the underlying libcloud client, so it's relatively straightforward to expose it to the salt-cloud driver.

I likely won't have time to address this straight away, but I'll make sure to follow up on this issue when it's ready for you to try out.

**Comment by rallytime** [comment]
Thanks for this report @fyatzeck and for jumping on this one @erjohnso!

**Comment by rallytime** [comment]
@fyatzeck Apologies for not coming back around to this earlier. I have tested the functionality of using a static ip with `external_ip` in 2015.8.0 with your example. (I suspect that #18650 may have had something to do with getting this working correctly, and that fix wasn't available until 2015.5.0), which would explain what you're seeing. It looks to be working well! :D

As for the ip-forwarding, that functionality was certainly missing. I have fixed this in #27350. This fix will be available in the upcoming 2015.8.1 release by adding `ip_forwarding: True` to your cloud config. Here is an example of the result of my fix:
```
[INFO    ] Created Cloud VM 'nt-gce-4'
nt-gce-4:
    ----------
    _uuid:
        None
    driver:
    extra:
        ----------
        canIpForward:
            True
<snipped>
```
